sup3r/preprocessing/data_handling/exo_extraction.py

```diff
@@ -0,0 +1,480 @@
+        return exo.data
+
+
+class TopoExtractH5(ExoExtract):
+    """TopoExtract for H5 files"""
+
+    @property
+    def source_data(self):
+        """Get the 1D array of elevation data from the exo_source_h5"""
+        with Resource(self._exo_source) as res:
+            elev = res.get_meta_arr('elevation')
+            elev = np.repeat(elev[:, np.newaxis], self.hr_shape[-1], axis=-1)
+        return elev
+
+    @property
+    def source_lat_lon(self):
+        """Get the 2D array (n, 2) of lat, lon data from the exo_source_h5"""
+        with Resource(self._exo_source) as res:
+            source_lat_lon = res.lat_lon
+        return source_lat_lon
+
+    @property
+    def source_time_index(self):
+        """Time index of the source exo data"""
+        if self._src_time_index is None:
+            with Resource(self._exo_source) as res:
+                self._src_time_index = res.time_index
+        return self._src_time_index
+
+    @property
+    def data(self):
+        """Get a raster of source values corresponding to the
+        high-resolution grid (the file_paths input grid * s_enhance *
+        t_enhance). The shape is (lats, lons, temporal, 1)
+        """
+        nn = self.nn
+        hr_data = []
+        for j in range(self._s_agg_factor):
+            out = self.source_data[nn[:, j]]
+            out = out.reshape(self.hr_shape)
+            hr_data.append(out[..., np.newaxis])
+        hr_data = np.concatenate(hr_data, axis=-1).mean(axis=-1)
+        logger.info('Finished mapping raster from {}'.format(self._exo_source))
+        return hr_data[..., np.newaxis]
+
+
+class TopoExtractNC(TopoExtractH5):
+    """TopoExtract for netCDF files"""
+
+    def __init__(self, *args, **kwargs):
+        """
+        Parameters
+        ----------
+        args : list
+            Same positional arguments as TopoExtract
+        kwargs : dict
+            Same keyword arguments as TopoExtract
+        """
+
+        super().__init__(*args, **kwargs)
+        logger.info('Getting topography for full domain from '
+                    f'{self._exo_source}')
+        self.source_handler = DataHandlerNC(
+            self._exo_source,
+            features=['topography'],
+            worker_kwargs=dict(ti_workers=self.ti_workers),
+            val_split=0.0,
+        )
+
+    @property
+    def source_data(self):
+        """Get the 1D array of elevation data from the exo_source_nc"""
+        elev = self.source_handler.data.reshape((-1, self.lr_shape[-1]))
+        return elev
+
+    @property
+    def source_lat_lon(self):
+        """Get the 2D array (n, 2) of lat, lon data from the exo_source_nc"""
+        source_lat_lon = self.source_handler.lat_lon.reshape((-1, 2))
+        return source_lat_lon
+
+
+class SzaExtract(ExoExtract):
+    """SzaExtract for H5 files"""
+
+    @property
+    def source_data(self):
+        """Get the 1D array of sza data from the exo_source_h5"""
+        return SolarPosition(self.hr_time_index,
```

Comment:
nice thanks for implementing this! Wait, no way this is a 1D array though, right?